frontend/openchat-shared/src/utils/immutable.ts

```diff
@@ -0,0 +1,17 @@
+export class Immutable<T> {
```

Comment:
This is a bit weird. Seems like it's purpose is not really to make something immutable but rather to clone it if necessary. Not sure of the purpose of `Object.freeze`. It's only a shallow freeze and it means that if we return the original we are returning a (partially) frozen version of it. Probably not usually harmful - but why?

Reply:
That `Object.freeze` can go, the return type is already `Readonly` so the compiler will do the same checks already.